go/chat/boxer.go

```diff
@@ -95,21 +100,24 @@ func (b *Boxer) UnboxMessage(ctx context.Context, finder *KeyFinder, boxed chat1
 
 	if matchKey == nil {
 		err := fmt.Errorf("no key found for generation %d", boxed.KeyGeneration)
-		return b.makeErrorMessage(boxed, err), libkb.ChatUnboxingError{Msg: err.Error()}
+		// transient error
+		return chat1.MessageUnboxed{}, libkb.NewChatUnboxingError(err.Error())
 	}
 
 	pt, headerHash, err := b.unboxMessageWithKey(ctx, boxed, matchKey)
 	if err != nil {
 		b.log().Warning("failed to unbox message: msgID: %d err: %s", boxed.ServerHeader.MessageID,
 			err.Error())
-		return b.makeErrorMessage(boxed, err), libkb.ChatUnboxingError{Msg: err.Error()}
+		// permanent error
+		return b.makeErrorMessage(boxed, err), nil
```

Comment:
oof, most of it looks static but `validSenderKey` can trigger `LoadUserByUID` through the `UserInfoMapper`. If that fails it should be a transient error. I wonder if that can ever happen.